### PR TITLE
fix(NcAppNavigationItem): use $scopedSlots instead of $slots

### PR DESCRIPTION
**Pull request by st3iny**
### ☑️ Resolves

- Fix https://github.com/nextcloud/mail/issues/10270

> Used to programmatically access [scoped slots](https://v2.vuejs.org/v2/guide/components#Scoped-Slots). For each slot, including the default one, the object contains a corresponding function that returns VNodes.

Ref https://v2.vuejs.org/v2/api/#vm-scopedSlots

#### Before
[Bildschirmaufnahme_20241022_194043.webm](https://github.com/user-attachments/assets/15a03f9e-1df2-48d6-9cb2-fc35527c8125)

#### After
[Bildschirmaufnahme_20241022_193652.webm](https://github.com/user-attachments/assets/ec91c380-b66d-4028-a987-e2f35657e177)

### 🏁 Checklist

- [ ] ⛑️ Tests are included or are not applicable
- [ ] 📘 Component documentation has been extended, updated or is not applicable
- [ ] 3️⃣ Backport to `next` requested with a Vue 3 upgrade

### CODE REVIEW

File: src/components/NcAppNavigationItem/NcAppNavigationItem.vue [hunk 1]
```diff
@@ -731,8 +731,8 @@ export default {
 		},
 
 		updateSlotInfo() {
-			this.hasChildren = !!this.$slots.default
-			this.collapsible = this.allowCollapse && !!this.$slots.default
+			this.hasChildren = !!this.$scopedSlots.default
+			this.collapsible = this.allowCollapse && !!this.$scopedSlots.default
```
[the remark this answers is not in the document]
Comment (st3iny) [reply src/components/NcAppNavigationItem/NcAppNavigationItem.vue:735]: I agree, the logic is a bit hacky. I refactored the code according to your recommendations and pushed another commit.

**Review by Antreesy — APPROVED** [review]
LGTM

**Review by susnux — APPROVED** [review]

**Comment by ShGKme** [comment]
Could you test that it still works dynamically generated content, when the list is empty?

**Comment by st3iny** [comment]
> Could you test that it still works dynamically generated content, when the list is empty?

What exactly do you mean with "dynamically generated content"? I tested it in Mail with my latest changes. The collapse toggle will be shown when there are children (and collapsing is allowed via the prop). Otherwise, it won't be shown.

**Comment by ShGKme** [comment]
> What exactly do you mean with "dynamically generated content"? I tested it in Mail with my latest changes. The collapse toggle will be shown when there are children (and collapsing is allowed via the prop). Otherwise, it won't be shown.

When the value of children dynamically changes from 0 to N and back without component re-mounting.

`$scopedSlots` and `$slots` are different, especially with the default slot. `$scopedSlots.default` may not be falsy, while `$scopedSlots.default()` is actually empty (renders no real content).

Tested on netlify that if works fine now.

**Pull request merged**